src/renderer/gdi/paint.cpp

```diff
@@ -41,6 +41,9 @@ using namespace Microsoft::Console::Render;
     _psInvalidData.hdc = GetDC(_hwndTargetWindow);
     RETURN_HR_IF_NULL(E_FAIL, _psInvalidData.hdc);
 
+    // We need the advanced graphics mode in order to set a transform.
+    SetGraphicsMode(_psInvalidData.hdc, GM_ADVANCED);

```

Comment:
> Does it impact the fidelity of the scaled text at all (is extra aliasing visible) or does it work perfectly as if you had set the font size larger?

It looks pretty smooth to me. It's definitely not just pixel-doubling. However, I should mention that I've noticed block characters not always lining up perfectly when double sized, but I'm not sure that's necessarily the fault of the transforms, and I didn't think it was worth worrying about too much until someone complained.

> it might cause all sorts of GDI stuff to take a different codepath that is slower in the common case

Yeah, I did wonder about that too. I'm not sure you can just turn it on and off as you need it though. And note that the renderer is now using transforms for the viewport offset too, so it's not just the double-size renditions that need it. But if it it's a significant problem, I can try and see what it would take to enable it on demand - I suspect it would be messy though.

Reply:
Nah based on the wpr... It didn't have a big impact just for setting the mode. So ignore this one.